### DISCUSSION TITLE
Highlighting important information

### DISCUSSION THREAD

**Discussion by elliot-butler-ho** [UKHomeOffice/design-system#455] [Ideas]
## What
A number of internal services across government have implemented a highlight pattern to focus a user's attention to particular parts of data. The pattern is used to visually highlight text, differentiating it from other information on the screen. It exists in the Home Office design system [figma file](https://www.figma.com/file/y90OmHLuhtNuCUOc7jzQlj/Home-Office-design-library-(Roboto)) and is used in Home Office services already: Entity search and CRS data breaches.

<img width="312" alt="image" src="https://user-images.githubusercontent.com/80390605/190617972-e02f62b9-32c9-457e-8e25-2fdb1da52b3d.png">

The following use cases of the pattern have been identified across gov so far:

- To highlight to users where data has changed - NHS cervical screening management system
- To highlight to users where data is different when comparing - Identity verification, Home Office
- To highlight to users where data matches when comparing - n'Delius records, MoJ
- To highlight to users matched data from a search request - Entity search, Home Office

## Why
The pattern has been well tested in multiple services and already exists on live services within Home Office and MoJ. 
The task to compare data is common, particularly for internal case working / management services. The ability to visually highlight elements should save the user time to find what they are looking for, especially when there is a lot of content on the page, or comparing multiple sets of data at once - valuable for repeat users. 

## Anything else

**Examples in existing services:**
Home Office Entity search - used to display matched search results
<img width="1508" alt="image" src="https://user-images.githubusercontent.com/80390605/190623236-edf0a75b-8ef3-4a4c-99f5-c4033f54e0cf.png">

(Proposed) mock up design to be used in Home Office UKVI identity verification - used to identify differences in two data sets
<img width="826" alt="image" src="https://user-images.githubusercontent.com/80390605/192783881-446b606e-b7fb-42c3-a5fd-755384969762.png">

NHS cervical screening management - used to identify changes in medical records
![image](https://user-images.githubusercontent.com/80390605/190612074-efe8c7d1-9177-4bf1-8248-a0fd03c66fdb.png)
![image](https://user-images.githubusercontent.com/80390605/190612173-8a63aece-aa28-4593-8469-4ff6f000828e.png)

MoJ - used to identity similar data in cases and merge them
![image](https://user-images.githubusercontent.com/80390605/190612303-6f2a6672-b0cf-4bfa-bcb4-11938663b0b1.png)

## Accessibility considerations
Even though we can call out the highlighted text for screen reader users, conscious that the visual colour may not be ideal for users who are partially sighted. Colour contrast against the Home Office grey background may also not be ideal. Curious to hear any comments around this.

**Comment by devansXD** [comment]
This will be relevant when thinking about accessibility https://service-manual.nhs.uk/design-system/styles/colour#accessibility

**Comment by elliot-butler-ho** [comment]
I also posted this thread on the [gov design system github](https://github.com/UKHomeOffice/design-system/discussions/455) and received some great insights / user feedback from NHS and MOJ services using the pattern. 

To summarise, the highlighted text has tested well with users, with positive feedback that it works well as a visual cue to draw attention to information.

Next steps will be to:
- review the accessibility side of the code and establish a recommended mark up
- review the shades of yellow to improve colour contrast for partially sighted users, particularly when used on a grey background, such as Home Office internal services.

**Comment by mrmartyndawson** [comment]
Entity search doesn't use this highlight pattern in the search results screen as above any more, this highlighted text is now a hyperlink through to a lower level details page.

As the contrast between the highlight (yellow) and the link colour (blue) fail accessibility guidelines, we've decided to remove the highlight.

**Comment by elliot-butler-ho** [comment]
Accessible code and semantics of the highlight pattern, made working with the accessibility and inclusion team - as tested in research sessions. 

Colour tints in CSS still to be finalised. 

HTML
`<mark class="ho-highlight"><strong>text here</strong></mark>`

CSS
```

.ho-highlight {
  background-color: #FFE5CC;
  border-bottom:2px solid #FFB266
}

mark::before,
mark::after {
  clip-path: inset(100%);
  clip: rect(1px, 1px, 1px, 1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

mark::before {
  content: " [highlight start] ";
}

mark::after {
  content: " [highlight end] ";
}
```

**Comment by tomyems** [comment]
Issue created for dev work on making the component: https://github.com/UKHomeOffice/design-system/issues/495

To discuss the design choice including colour or guidance for the page, use this discussion.
To discuss the technical implementation including writing tests and so on, use the issue.

**Comment by tomyems** [comment]
Improved strength of research findings: https://63fcabeb767019005583b658--hods.netlify.app/components?name=Highlight

**Reply by tomyems** [reply]
Rephrased to 'severities of vision loss: https://63fcbae52b356b0d85c997c7--hods.netlify.app/components?name=Highlight

**Reply by tomyems** [reply]
Corrected services using this components: https://6407506a6a15180f524def75--hods.netlify.app/components?name=Highlight

**Comment by tomyems** [comment]
Add to guidance an explanation of how this interacts with screen reader users:

> when there’s some highlighted text it’ll say “highlight start” -( then reads the highlighted text) followed by “highlight end”  where the highlight has stopped

**Comment by millie-drs** [comment]
We are using the highlight and compare pattern but with slight amendments. 
-Not bold text within the highlight
-Comparison underneath (instead of side-by-side)
-Include a ‘summary’ at the top of the page to let users know that there are highlights further down the page 

**Rationale:**
- One type of information takes precedence over the other, so needed more hierarchy in the information. Therefore, we only wanted to highlight the secondary and not the primary.
- Side by side didn’t scale well for smaller screens – the 2 sets of tabled data sit one above the other so that the compared information is quite separated on a smaller screen.
- Bold highlight not appropriate because it’s secondary information. Also, we had some where the highlight may be needed in the middle of a line of text, so bold felt off balance. Equally we made the labels for the secondary information not bold, so that it was easier on the eye and the two label categories could be compared.

The other amendment we made was to include a summary, of a similar template to the gds error summary, but using the colours of the highlight. The information may not be incorrect that is being highlighted so it was inappropriate to use a red error pattern. We used the same tinted orange colour-way to tie them visually together. 

Initial results show that these tested well. We are continuing to test but sharing our early findings here.

<img width="575" height="461" alt="Screenshot 2026-06-24 at 17 23 45" src="https://github.com/user-attachments/assets/c15b6194-88eb-4c70-b9e7-61f1371960b8" />
<img width="575" height="494" alt="Screenshot 2026-06-24 at 17 21 33" src="https://github.com/user-attachments/assets/d8f4901c-6d39-4a66-95a4-b05d8fb977a2" />
